packages/sanity/src/core/releases/tool/components/ReleaseMenuButton/ReleaseMenuButton.tsx

```diff
@@ -17,127 +18,228 @@ export type ReleaseMenuButtonProps = {
   release: ReleaseDocument
 }
 
-const ARCHIVABLE_STATES = ['active', 'published']
+type ReleaseAction = 'archive' | 'delete'
+
+interface ReleaseActionMap {
+  dialogId: string
+  dialogHeaderI18nKey: string
+  dialogDescriptionSingularI18nKey: string
+  dialogDescriptionMultipleI18nKey: string
+  dialogConfirmButtonI18nKey: string
+  toastSuccessI18nKey: string
+  toastFailureI18nKey: string
+  telemetry: DefinedTelemetryLog<void>
+}
+
+const RELEASE_ACTION_MAP: Record<ReleaseAction, ReleaseActionMap> = {
```

Comment:
I expect that unarchive will follow much the same approach as these 2 actions, so this can be easily extended for that